caddyhttp/rewrite/to.go

```diff
@@ -49,7 +51,7 @@ func To(fs http.FileSystem, r *http.Request, to string, replacer httpserver.Repl
 
 	// take note of this rewrite for internal use by fastcgi
 	// all we need is the URI, not full URL
-	r.Header.Set(headerFieldName, r.URL.RequestURI())
+	*r = *r.WithContext(context.WithValue(r.Context(), caddy.URIxRewriteCtxKey, r.URL.RequestURI()))
```

Comment:
This was the bit that gave me a little bit of a headache, but essentially if I just added context to r since this call `To` just returns and doesnt not pass the request further down (if that makes sense), the context that I had added was not available outside the function unless I dereferenced r.